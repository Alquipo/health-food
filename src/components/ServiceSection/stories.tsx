import { Story, Meta } from '@storybook/react'
import ServiceSection from '.'

export default {
  title: 'ServiceSection',
  component: ServiceSection
} as Meta

export const Default: Story = () => <ServiceSection />
