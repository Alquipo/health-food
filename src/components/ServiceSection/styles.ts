import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.section`
  ${({ theme }) => css`
    display: flex;
    max-height: 70rem;
    ${media.lessThan('large')`
      padding: ${theme.spacings.xxlarge} ${theme.spacings.xsmall};
    `};
  `}
`

export const Image = styled.img`
  flex: 1;
  object-fit: cover;

  ${media.lessThan('large')`
    display: none;
  `};
`

export const Content = styled.div`
  flex: 1;
  display: flex;
  align-items: center;
  width: 100%;

  ${media.lessThan('large')`
    align-items: center;
    justify-content: center;
  `};
`

export const TextWrapper = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: 55rem;

    button {
      align-self: flex-start;
      margin-top: ${theme.spacings.medium};
    }
  `}
`

export const Title = styled.h2`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.medium};
    font-size: ${theme.font.sizes.huge};
    color: ${theme.colors.secondary};
    letter-spacing: 0.1rem;
    line-height: 1.3;

    ${media.lessThan('small')`
      font-size: ${theme.font.sizes.xxxlarge};

    `};
  `}
`

export const Text = styled.p`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.small};
    color: ${theme.colors.text};
    line-height: 1.6;
    font-size: ${theme.font.sizes.xlarge};

    ${media.lessThan('small')`
      font-size: ${theme.font.sizes.large};

    `};
  `}
`
