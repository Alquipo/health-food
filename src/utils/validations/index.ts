import * as yup from 'yup'

const nascimentoRegex =
  /^((((19|[2-9]\d)\d{2})-(0[13578]|1[02])-(0[1-9]|[12]\d|3[01]))|(((19|[2-9]\d)\d{2})-(0[13456789]|1[012])-(0[1-9]|[12]\d|30))|(((19|[2-9]\d)\d{2})-02-(0[1-9]|1\d|2[0-8]))|(((1[6-9]|[2-9]\d)(0[48]|[2468][048]|[13579][26])|((16|[2468][048]|[3579][26])00))-02-29))$/g

const cpfRegex = /^\d{3}.?\d{3}.?\d{3}-?\d{2}$/

const cepRegex = /^([\d]{2})\.*([\d]{3})-*([\d]{3})/

const estadoRegex =
  /^(AC|AL|AP|AM|BA|CE|DF|ES|GO|MA|MT|MS|MG|PA|PB|PR|PE|PI|RJ|RN|RS|RO|RR|SC|SP|SE|TO)$/i

export const fieldsValidationSchema = yup.object({
  name: yup.string().required('Is not allowed to be empty'),
  birthDate: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(nascimentoRegex, 'must be a valid email'),
  cpf: yup
    .string()
    .max(14, 'Must be 14 characters')
    .required('Is not allowed to be empty')
    .matches(cpfRegex, 'Invalid CPF'),
  zipCode: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(cepRegex, 'Invalid ZipCode'),
  address: yup.string().required('Is not allowed to be empty'),
  district: yup.string().required('Is not allowed to be empty'),
  number: yup.number().required('Is not allowed to be empty'),
  city: yup.string().required('Is not allowed to be empty'),
  state: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(estadoRegex, 'Invalid State')
})
