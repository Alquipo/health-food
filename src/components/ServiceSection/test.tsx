import theme from 'styles/theme'
import { render, screen } from 'utils/test-utils'

import ServiceSection from '.'

describe('<ServiceSection/>', () => {
  it('should render a service section', () => {
    const { container } = render(<ServiceSection />)

    expect(
      screen.getByRole('heading', {
        name: /The best services ready To serve you/i
      })
    ).toBeInTheDocument()

    expect(
      screen.getByText(
        /Far far away, behind the word mountains, far from the countries Vokalia and Consonantia, there live the blind texts/i
      )
    ).toBeInTheDocument()

    expect(
      screen.getByText(
        /Separated they live in Bookmarksgrove right at the coast of the Semantics, a large language ocean./i
      )
    ).toBeInTheDocument()

    expect(
      screen.getByText(
        /A small river named Duden flows by their place and supplies it with the necessary regelialia./i
      )
    ).toBeInTheDocument()

    expect(screen.getByRole('button', { name: /Know More/i })).toHaveStyleRule(
      'box-shadow',
      `0 1.3rem 2.1rem -0.8rem ${theme.colors.primary}`
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
