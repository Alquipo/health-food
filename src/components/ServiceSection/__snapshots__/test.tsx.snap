// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServiceSection/> should render a service section 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-height: 70rem;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  object-fit: cover;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c3 {
  width: 100%;
  max-width: 55rem;
}

.c3 button {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin-top: 3.2rem;
}

.c4 {
  margin-bottom: 3.2rem;
  font-size: 4.8rem;
  color: #1D164D;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  line-height: 1.3;
}

.c5 {
  margin-top: 2.4rem;
  color: #9E9BAF;
  line-height: 1.6;
  font-size: 2.0rem;
}

.c6 {
  padding: 1.6rem 2.4rem;
  background: #BADC58;
  color: #FFFFFF;
  border-radius: 0.6rem;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
  box-shadow: 0 1.3rem 2.1rem -0.8rem #BADC58;
  cursor: pointer;
  border: none;
  font-size: 1.8rem;
  font-weight: 600;
}

.c6:hover {
  -webkit-filter: brightness(0.95);
  filter: brightness(0.95);
}

@media (max-width:1170px) {
  .c0 {
    padding: 5.6rem 1.6rem;
  }
}

@media (max-width:1170px) {
  .c1 {
    display: none;
  }
}

@media (max-width:1170px) {
  .c2 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:450px) {
  .c4 {
    font-size: 3.6rem;
  }
}

@media (max-width:450px) {
  .c5 {
    font-size: 1.8rem;
  }
}

<section
  class="c0"
>
  <img
    class="c1"
    src="/img/bloco_services.svg"
  />
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <h2
        class="c4"
      >
        The best services ready To serve you
      </h2>
      <p
        class="c5"
      >
        Far far away, behind the word mountains, far from the countries Vokalia and Consonantia, there live the blind texts
      </p>
      <p
        class="c5"
      >
        Separated they live in Bookmarksgrove right at the coast of the Semantics, a large language ocean.
      </p>
      <p
        class="c5"
      >
        A small river named Duden flows by their place and supplies it with the necessary regelialia.
      </p>
      <button
        class="c6"
        type="button"
      >
        Know More
      </button>
    </div>
  </div>
</section>
`;
